lib/improver/utilities/spatial.py

```diff
@@ -0,0 +1,163 @@
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+""" Provides support utilities."""
+
+import copy
+from iris.coords import CellMethod, DimCoord
+from iris.cube import Cube
+from iris.exceptions import InvalidCubeError
+import numpy as np
+
+
+class DiscreteDifferenceBetweenAdjacentGridSquares(object):
+
+    """
+    Calculate the discrete difference between adjacent grid squares within
+    a cube. The discrete difference is calculated along the x and y axis
+    individually.
+    """
+
+    def __init__(self):
+        """
+        Initialise class.
+        """
+        pass
+
+    @staticmethod
+    def create_discrete_difference_cube(
+            cube, coord_name, diffs_along_axis):
+        """
+        Put the discrete difference array into a cube with the appropriate
+        metadata.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube from which the discrete differences have been calculated.
+        coord_name : String
+            The name of the coordinate over which the discrete difference
+            have been calculated.
+        diffs_along_axis : numpy array
+            Array containing the discrete differences.
+
+        Returns
+        -------
+        discrete_diff_cube : Iris.cube.Cube
+            Cube containing the discrete differences calculated along the
+            specified axis.
+        """
+        points = cube.coord(coord_name).points
+        mean_points = (points[1:] + points[:-1]) / 2
+
+        # Copy cube metadata and coordinates into a new cube.
+        # Create a new coordinate for the coordinate along which the discrete
+        # difference has been calculated.
+        metadata_dict = copy.deepcopy(cube.metadata._asdict())
+        discrete_diff_cube = Cube(diffs_along_axis, **metadata_dict)
+
+        for coord in cube.dim_coords:
+            dims = cube.coord_dims(coord)
+            if coord.name() in [coord_name]:
+                coord = DimCoord(
+                    mean_points, standard_name=coord.standard_name,
+                    units=coord.units)
+            discrete_diff_cube.add_dim_coord(coord.copy(), dims)
+        for coord in cube.aux_coords:
+            dims = cube.coord_dims(coord)
+            discrete_diff_cube.add_aux_coord(coord.copy(), dims)
+        for coord in cube.derived_coords:
+            dims = cube.coord_dims(coord)
+            discrete_diff_cube.add_aux_coord(coord.copy(), dims)
+
+        # Add metadata to indicate that a discrete difference has been
+        # calculated.
+        cell_method = CellMethod("discrete_difference", coords=[coord_name])
+        discrete_diff_cube.add_cell_method(cell_method)
+        discrete_diff_cube.attributes["form_of_discrete_difference"] = (
+            "forward_difference")
```

Comment:
After discussion, we've settled on these rules temporarily:
* Remove 'discrete' from all class, function, variable names as the quantity is really a difference over the grid length.
* Keep the cell_method, add the interval (1 grid length).
* Keep the input cube units. Iris won't recognise '1 grid length' as a unit.
A `TODO` has been added to the `create_difference_cube` method to remind developers this metadata may need to be changed when more formal conventions are agreed upon.